WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift

```diff
@@ -706,42 +706,49 @@ extension NotificationsViewController {
 
         // Display Details
         //
-        if let postID = note.metaPostID, let siteID = note.metaSiteID, note.kind == .matcher || note.kind == .newPost {
+        if let postID = note.metaPostID,
+            let siteID = note.metaSiteID,
+            note.kind == .matcher || note.kind == .newPost {
             let readerViewController = ReaderDetailViewController.controllerWithPostID(postID, siteID: siteID)
             readerViewController.navigationItem.largeTitleDisplayMode = .never
             showDetailViewController(readerViewController, sender: nil)
 
             return
         }
 
+        presentCommentDetail(for: note)
+    }
+
+    private func presentCommentDetail(for note: Notification) {
         // This dispatch avoids a bug that was occurring occasionally where navigation (nav bar and tab bar)
         // would be missing entirely when launching the app from the background and presenting a notification.
         // The issue seems tied to performing a `pop` in `prepareToShowDetails` and presenting
         // the new detail view controller at the same time. More info: https://github.com/wordpress-mobile/WordPress-iOS/issues/6976
         //
         // Plus: Avoid pushing multiple DetailsViewController's, upon quick & repeated touch events.
-        //
+
         view.isUserInteractionEnabled = false
 
         DispatchQueue.main.async {
             if FeatureFlag.notificationCommentDetails.enabled,
                note.kind == .comment {
                 let notificationCommentDetailCoordinator = NotificationCommentDetailCoordinator(notification: note)
 
-                // For now, NotificationCommentDetailCoordinator only loads the Comment if it is cached.
-                // If the comment is not cached, fall back to showing the old comment view.
-                // This is temporary until NotificationCommentDetailCoordinator can fetch the comment from the endpoint.
+                notificationCommentDetailCoordinator.createViewController { commentDetailViewController in
+                    guard let commentDetailViewController = commentDetailViewController else {
+                        // TODO: show error view
+                        return
```

Comment:
When navigating into a comment view I'm seeing first one comment screen and then the other pushed on, one after the other:

https://user-images.githubusercontent.com/4780/152567230-47b789a1-6910-465f-abc9-b23b1c39e028.mov

I think this is because the return statements in this block only return from the block, they don't return from the `presentCommentDetail` method or stop it executing. I think one fix would be to put the `performSegue` below in an `else` statement.

Reply:
Well, that be because I put the `return` in the wrong place. And clearly forgot to test it. 🤦 Sorry about that! It should be fixed now.